backend/yonder/views.py

```diff
@@ -176,7 +176,43 @@ class post_detail(generics.RetrieveUpdateDestroyAPIView):
 
     @swagger_auto_schema(tags=['posts'])
     def get(self, request, *args, **kwargs):
-        return self.retrieve(request, *args, **kwargs)
+        try:  
+            post = Post.objects.get(id=self.kwargs["pk"])
+            postJSON = PostSerializer(instance=post).data
+            if post.visibility == "FRIENDS":
+                req_user = User.objects.get(username=request.user)
+                requestor = Author.objects.get(user=req_user)
+                follows = AuthorFollower.objects.filter(author_id=post.author.id)
+                for follow in follows:
+                    if follow.follower["id"] == str(requestor.id):
+                        _follows = AuthorFollower.objects.filter(author_id=requestor.id)
+                        for _follow in _follows:
+                            if _follow.follower["id"] == self.kwargs["author_id"]:
+                                return Response(postJSON, status=status.HTTP_200_OK)
+
+                return Response(status=status.HTTP_401_UNAUTHORIZED)
+
+            elif post.visibility == "PUBLIC":
+                return Response(postJSON, status=status.HTTP_200_OK)
+
+            else:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        except Post.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        except Author.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        except AuthorFollower.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        
+        except User.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        except Exception as e:
+            print(e)
+            return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
Instead of all the exceptions use `get_object_or_404` rather than `objects.get`. That'll much cleaner and there would be no need for a `try` statement nor returning a 500.